### DISCUSSION TITLE
Freezing configuration

### DISCUSSION THREAD

**Discussion by rkennedy-mode** [duckdb/duckdb#7649] [Ideas]
I'd like to propose a mechanism to "freeze" the DuckDB [configuration](https://duckdb.org/docs/sql/configuration). This would allow hosted environments to configure DuckDB and then lock it in a way that prevents users from making further changes to any configuration value.

`enable_external_access` appears to prevent changes once it's been disabled:

```
D select count(*) from 'hits.csv';
┌──────────────┐
│ count_star() │
│    int64     │
├──────────────┤
│     99997497 │
└──────────────┘
D set enable_external_access = false;
D select count(*) from 'hits.csv';
Error: Permission Error: Scanning CSV files is disabled through configuration
D set enable_external_access = true;
Error: Invalid Input Error: Cannot change enable_external_access setting while database is running
```

However, this feature of `enable_external_access` appears to be undocumented and none of the other configuration values appear to make any statements regarding their mutability once set. In addition, as new configuration values are added it becomes important for hosting platforms to remember to set any newly-introduced sensitive values.

In order to make this easier for hosting platforms, DuckDB could introduce some sort of directive that could be used after the hosted platform's desired settings have been applied, which prevents future changes/resets to any configuration values.

**Comment by Mytherin** [comment]
That sounds like a good idea to me - I have turned it into an issue here #7662. Let me know if I missed anything.

**Reply by rkennedy-mode** [reply]
That looks perfect. Thanks, @Mytherin!
